tests/unit/test_caching.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+
+from opendevin.core.config import LLMConfig
+from opendevin.llm.llm import LLM
+
+sample_text = """
```

Comment:
I love that you set up tests! But wait, we don't run this in CI on Anthropic. Maybe we can mock here just to see that we send the right things, and for the rest it'll be something for our local machines? Cc: @li-boxuan

Reply:
I'd say the best we could do is to have some unittests checking the headers.